### PR TITLE
Add --interactive option to sync, delete and destroy / Remove --interactive from global options

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

1. add interactive in sync 
2. remove --interactive in global options

not all subcommand has interactive mode.

Fixed: https://github.com/helmfile/helmfile/issues/324

@mumoshu

### CODE REVIEW

**Comment by yxxhero** [comment]
@mumoshu Ready to review. WDYT?

File: pkg/app/app.go [hunk 1]
```diff
@@ -1661,9 +1661,16 @@ func (a *App) sync(r *Run, c SyncConfigProvider) (bool, []error) {
 
 	infoMsg := fmt.Sprintf(`Affected releases are:
 %s
+
+Do you really want to sync?
+  Helmfile will sync all your releases, as shown above.
```
Comment (mumoshu) [inline pkg/app/app.go:1666]: Seeing how it works for `apply` today, this question is printed only when the interactive mode is enabled.

Almost certainly we should follow the same pattern?

The key difference is, in `apply`, we have two variables- one for `infoMsg` and another for `confMsg` and we print `infoMsg` for the non-interactive case and `confMsg` for the interactive case.

Comment (yxxhero) [reply pkg/app/app.go:1666]: ok. I will see it.

**Comment by yxxhero** [comment]
@mumoshu done
1. add interactive in sync 
2. remove --interactive in global options
3. in cmd/apply.go applyImpl.ApplyOptions --> applyOptions reduce the code.

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks for enhancing it!

**Pull request merged**